frontend/src/metabase/actions/components/ActionViz/Action.tsx

```diff
@@ -84,13 +84,13 @@ export function ActionComponent({
 
   const onSubmit = useCallback(
     (parameterMap: ParametersForActionExecution) => {
+      const action = dashcard.action;
+      const fieldSettings =
+        action?.visualization_settings?.fields ||
+        generateFieldSettingsFromParameters(action?.parameters ?? []);
+
       const params = {
-        ...setNumericValues(
-          dashcardParamValues,
-          generateFieldSettingsFromParameters(
-            dashcard?.action?.parameters ?? [],
-          ),
-        ),
+        ...setNumericValues(dashcardParamValues, fieldSettings),
         ...parameterMap,
       };
```

Comment:
I think this little change will make this PR also close https://github.com/metabase/metabase/issues/28953

Essentially doing the same coercion based on settings for dashboard and form values.

```suggestion
        ...setNumericValues({
          ...dashcardParamValues, 
          ...parameterMap,
         }, fieldSettings),
      };
```

Reply:
@iethree you're absolutely right! If you don't mind, I'd do this in another PR because we also want to validate numeric dropdown values + add some test coverage around all that